File: mobile_app/lib/classes/transfer.dart
import 'package:intl/intl.dart';

class Transfer {
  final String firstPlateDate;
  final String plateDate;
  final String startTranferDate;
  final String writeTransferDate;
  final String endTransferDate;
  final String typeTransfer;
  final String zipCode;
  final String person;
  final String typeServiceVehicle;
  final String city;

  Transfer({
    required this.firstPlateDate,
    required this.plateDate,
    required this.startTranferDate,
    required this.writeTransferDate,
    required this.endTransferDate,
    required this.typeTransfer,
    required this.zipCode,
    required this.person,
    required this.typeServiceVehicle,
    required this.city,
  });

  factory Transfer.fromJson(Map<String, dynamic> json) {
    var values = json.values.toList();
    var dateFormat = DateFormat('yyyy/MM/dd');

    return Transfer(
      firstPlateDate:
          values[0] != null ? dateFormat.format(DateTime.parse(values[0])) : '',
      plateDate:
          values[1] != null ? dateFormat.format(DateTime.parse(values[1])) : '',
      startTranferDate:
          values[2] != null ? dateFormat.format(DateTime.parse(values[2])) : '',
      writeTransferDate:
          values[3] != null ? dateFormat.format(DateTime.parse(values[3])) : '',
      endTransferDate:
          values[4] != null ? dateFormat.format(DateTime.parse(values[4])) : '',
      typeTransfer: values[5].toString(),
      zipCode: values[6].toString(),
      person: values[7].toString(),
      typeServiceVehicle: values[8].toString(),
      city: values[9].toString(),
    );
  }
}

File: backend_v3/src/v1/routes/transfer.ts
import express from 'express'
import * as transferController from '../../controllers/transferController'

const router = express.Router()

/**
 * @openapi
 * /api/v1/transfers/{vehicleId}:
 *   get:
 *     summary: Get a vehicle's transfers
 *     parameters:
 *       - in: path
 *         name: vehicleId
 *         schema:
 *           type: string
 *         required: true
 *         description: Vehicle UUID
 *         example: 12d1d060-c025-11ec-9a04-005056101f82
 *     tags:
 *       - Transfers
 *     responses:
 *       200:
 *         description: Get a vehicle's transfers
 *         content:
 *           application/json:
 *             schema:
 *               type: object
 *               properties:
 *                 status:
 *                   type: string
 *                 data:
 *                   type: array
 *                   items:
 *                    type: object
 *                    properties:
 *                      startTransferDate:
 *                        type: string
 *                      writeTransferDate:
 *                        type: string
 *                      endTransferDate:
 *                        type: string
 *                      typeTransfer:
 *                        type: string
 *                      zipCode:
 *                        type: number
 *                      person:
 *                        type: string
 *                      typeServiceVehicle:
 *                        type: string
 *                      city:
 *                        type: string
 *       400:
 *         description: Bad request
 *         content:
 *           application/json:
 *             schema:
 *               type: object
 *               properties:
 *                 status:
 *                  type: string
 *                 info:
 *                  type: string
 *               example:
 *                 - status: "KO"
 *                   info: "Invalid id format 🙄"
 */
router.get('/:id', transferController.checkValidInput, transferController.getTransfersFromVehicle)

router.get('/:id/details', (_, res) => {
  res.send('Work in progress')
})

export default router
